### PR TITLE
Change type of metrics enabled to boolean

### PR DESCRIPTION
**Pull request by walle**
Changes the metrics.enabled configuration from string to boolean.

### CODE REVIEW

File: controllers/storageoscluster/controller.go [hunk 1]
```diff
@@ -118,8 +118,8 @@ func (c *StorageOSClusterController) UpdateStatus(ctx context.Context, obj clien
 	csiCondition := getCSICondition(ctx, c.Client, obj.GetNamespace(), log)
 	meta.SetStatusCondition(&cluster.Status.Conditions, csiCondition)
 
-	metricsEnabled, err := AreMetricsEnabled(cluster)
-	if err != nil || !metricsEnabled {
+	metricsEnabled := cluster.Spec.Metrics.Enabled
+	if !metricsEnabled {
```
Comment (Ricardo-Osorio) [inline controllers/storageoscluster/controller.go:122]: No need for that extra step/variable IMO
```suggestion
	if !cluster.Spec.Metrics.Enabled {
```

File: api/v1/storageoscluster_types.go [hunk 2]
```diff
@@ -230,6 +231,11 @@ type StorageOSClusterKVBackend struct {
 	Backend string `json:"backend,omitempty"`
 }
 
+// PVCMetrics contains the configuration for the PVC metrics.
+type PVCMetrics struct {
```
Comment (Mojachieee) [inline api/v1/storageoscluster_types.go:235]: nit: I would just name this `Metrics`. It's likely we'll have lots of different metrics here, not just metrics for PVCs 

Please ignore me if @Ricardo-Osorio has suggested something better \:)

**Review by Mojachieee — APPROVED** [review]

File: api/v1/storageoscluster_types.go (hunk 2 above)
Comment (Ricardo-Osorio) [reply api/v1/storageoscluster_types.go:235]: No, that's a good point 👍

Comment (walle) [reply api/v1/storageoscluster_types.go:235]: Discussed the naming with @Ricardo-Osorio beforehand, so will ignore :)

**Pull request merged**